src/compile/typecheck.rs

```diff
@@ -646,6 +654,7 @@ impl AbstractSyntaxTree for TypeCheckedExpr {
                         .flatten(),
                 )
                 .collect(),
+            TypeCheckedExprKind::IfLet(_, _, _, _, _) => unimplemented!(),
```

Comment:
Need a comment here to explain what's up.  Is this something that needs to be implemented, before if let expressions can be used?  Or is this something that should never be called, for some reason that can be explained?

Reply:
This logic is currently only used by the inlining code, so I didn't notice it still needed to be implemented when I submitted this PR.

Fixing this will only need a couple lines so expect to see a fix soon.